src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt

```diff
@@ -751,12 +749,16 @@ internal open class AccountSupervisor(
         }
     }
 
-    internal open fun triggerCompliance(action: ComplianceAction, callback: TransactionCallback) {
+    internal open fun triggerCompliance(action: ComplianceAction, callback: TransactionCallback?) {
         if (compliance.status != ComplianceStatus.UNKNOWN) {
             updateCompliance(DydxAddress(accountAddress), compliance.status, action)
-            callback(true, null, null)
+            if (callback != null) {
+                callback(true, null, null)
+            }
+        }
+        if (callback != null) {
+            callback(false, V4TransactionErrors.error(null, "No account address"), null)
```

Comment:
[discussion] i feel like since we can't use named args for passed in callback functions, it might be nice in the future to create a `{functionName}Args` dataclass so the fn signatures remain explicit 🤔